vyper/venom/passes/remove_unused_variables.py

```diff
@@ -11,9 +12,19 @@ class RemoveUnusedVariablesPass(IRPass):
 
     dfg: DFGAnalysis
     work_list: OrderedSet[IRInstruction]
+    reads_msize: bool
 
     def run_pass(self):
         self.dfg = self.analyses_cache.request_analysis(DFGAnalysis)
+        self.reachable = self.analyses_cache.request_analysis(ReachableAnalysis).reachable
+
+        self.reads_msize = {}
+        self.instruction_index = {}
+        for bb in self.function.get_basic_blocks():
+            for idx, inst in enumerate(bb.instructions):
+                self.instruction_index[inst] = idx
+                if inst.opcode == "msize" and bb not in self.reads_msize:
+                    self.reads_msize[bb] = idx
```

Comment:
Should not this store last `msize` in basic block instead of first `msize`. This could be the problem in case which is showed in PR https://github.com/charles-cooper/vyper/pull/53